### PR TITLE
MINOR: Remove deprecated KTable#writeAs, print, foreach, to, through

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by bbejeck — APPROVED** [review]
Thanks, @guozhangwang LGTM

**Comment by guozhangwang** [comment]
@bbejeck @vvcephei @mjsax this PR is ready for reviews.

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableImplTest.java [hunk 1]
```diff
@@ -103,7 +104,8 @@ public boolean test(String key, Integer value) {
 
         table3.toStream().process(supplier);
 
-        KTable<String, String> table4 = table1.through(stringSerde, stringSerde, topic2, storeName2);
+        table1.toStream().to(topic2, produced);
+        KTable<String, String> table4 = builder.table(topic2, consumed);
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableImplTest.java]: nit: final

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableImplTest.java [hunk 2]
```diff
@@ -150,8 +151,9 @@ public boolean test(String key, Integer value) {
                         return (value % 2) == 0;
                     }
                 });
-        KTableImpl<String, String, String> table4 = (KTableImpl<String, String, String>)
-                table1.through(stringSerde, stringSerde, topic2, storeName2);
+
+        table1.toStream().to(topic2, produced);
+        KTableImpl<String, String, String> table4 = (KTableImpl<String, String, String>) builder.table(topic2, consumed);
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableImplTest.java]: nit: final

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableMapValuesTest.java [hunk 3]
```diff
@@ -231,8 +232,8 @@ public boolean test(String key, Integer value) {
                         return (value % 2) == 0;
                     }
                 });
-        KTableImpl<String, String, String> table4 = (KTableImpl<String, String, String>)
-                table1.through(stringSerde, stringSerde, topic2, storeName2);
+        table1.toStream().to(topic2, produced);
+        KTableImpl<String, String, String> table4 = (KTableImpl<String, String, String>) builder.table(topic2, consumed);
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableMapValuesTest.java]: nit: final

**Review by bbejeck — APPROVED** [review]
Thanks, @guozhangwang. Just a few nits, LGTM

File: streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java [hunk 4]
```diff
@@ -168,6 +168,7 @@ public void shouldReduce() throws Exception {
         produceMessages(mockTime.milliseconds());
         groupedStream
             .reduce(reducer, "reduce-by-key")
+            .toStream()
             .to(Serdes.String(), Serdes.String(), outputTopic);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java:172]: as above

File: streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java [hunk 5]
```diff
@@ -294,6 +295,7 @@ public void shouldAggregate() throws Exception {
             aggregator,
             Serdes.Integer(),
             "aggregate-by-selected-key")
+            .toStream()
             .to(Serdes.String(), Serdes.Integer(), outputTopic);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java:299]: as above

File: streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationDedupIntegrationTest.java [hunk 6]
```diff
@@ -126,9 +126,9 @@ public void whenShuttingDown() throws IOException {
     @Test
     public void shouldReduce() throws Exception {
         produceMessages(System.currentTimeMillis());
-        groupedStream
-            .reduce(reducer, "reduce-by-key")
-            .to(Serdes.String(), Serdes.String(), outputTopic);
+        groupedStream.reduce(reducer, "reduce-by-key")
+                .toStream()
+                .to(Serdes.String(), Serdes.String(), outputTopic);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationDedupIntegrationTest.java:131]: Shouldn't this be `to(outputTopic, Produced.with(...))` ?

File: streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java [hunk 7]
```diff
@@ -445,7 +447,8 @@ public void shouldCount() throws Exception {
         produceMessages(mockTime.milliseconds());
 
         groupedStream.count("count-by-key")
-            .to(Serdes.String(), Serdes.Long(), outputTopic);
+                .toStream()
+                .to(Serdes.String(), Serdes.Long(), outputTopic);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java:451]: as above

File: streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java [hunk 8]
```diff
@@ -456,7 +459,8 @@ public void shouldCountWithInternalStore() throws Exception {
         produceMessages(mockTime.milliseconds());
 
         groupedStream.count()
-            .to(Serdes.String(), Serdes.Long(), outputTopic);
+                .toStream()
+                .to(Serdes.String(), Serdes.Long(), outputTopic);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java:463]: as above

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableMapValuesTest.java [hunk 9]
```diff
@@ -261,8 +261,8 @@ public boolean test(String key, Integer value) {
                     return (value % 2) == 0;
                 }
             }, Materialized.<String, Integer, KeyValueStore<Bytes, byte[]>>as("anyFilterName").withValueSerde(Serdes.Integer()));
-        KTableImpl<String, String, String> table4 = (KTableImpl<String, String, String>)
-            table1.through(stringSerde, stringSerde, topic2, storeName2);
+        table1.toStream().to(topic2, produced);
+        KTableImpl<String, String, String> table4 = (KTableImpl<String, String, String>) builder.table(topic2, consumed);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableMapValuesTest.java]: nit: `final`

**Review by mjsax — COMMENTED** [review]
Overall LGTM.

File: streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationDedupIntegrationTest.java (hunk 6 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationDedupIntegrationTest.java:131]: I plan to remove the part related to KIP-182 in another PR. This is just for KTable's deprecated ones.

**Review by mjsax — APPROVED** [review]
LGTM.

**Pull request merged**